input/NumberInput.qml

```diff
@@ -13,7 +15,23 @@ BaseInput {
 	onMaxChanged: { this.element.setProperty('max', value) }
 	onStepChanged: { this.element.setProperty('step', value) }
 	onValueChanged: {
-		this.value = this._setValueWithLimits(value);
+		this.startDebounce(value);
+	}
+
+	Timer {
+		id: debounceTimer;
+		interval: rootNumberInput.debounceTimeout;
+		property float value;
+		triggeredOnStart: false;
+
+		onTriggered: {
+			rootNumberInput.value = rootNumberInput._setValueWithLimits(this.value);
```

Comment:
we're trying not to introduce ids in components if it's possible, could you use 
this.parent._setValueWithLimits please?

Reply:
Removed root Id.  Simplified a bit more.  Couldn't get rid of the debounceTimer Id without difficulty so left it since I noticed that BaseInput already has a focuseTimer Id.